### PR TITLE
Add github license integration

### PR DESCRIPTION
**Pull request by montanaflynn**
Added a license badge using the new [GitHub license API](https://developer.github.com/v3/licenses/#get-a-repositorys-license). Not sure what happened to #280 & #315 but the badge wasn't available in master and using the official API seems to be a better solution.

### CODE REVIEW

**Comment by ahmadnassri** [comment]
nice :+1:

File: server.js [hunk 1]
```diff
@@ -2113,6 +2113,54 @@ cache(function(data, match, sendBadge, request) {
...
+  request(apiUrl, { headers: customHeaders }, function(err, res, buffer) {
+    if (err != null) {
+      badgeData.text[1] = 'inaccessible';
+      sendBadge(format, badgeData);
+    }
+    try {
+      if ((+res.headers['x-ratelimit-remaining']) === 0) {
+        return;  // Hope for the best in the cache.
+      } else if (res.statusCode === 404) {
+        throw new Error(); // Show invalid badge in this case
```
Comment (espadrine) [inline server.js]: We shouldn't throw unless our code is wrong or the server should shut down.

```js
badgeData.text[1] = 'not found';
sendBadge(format, badgeData);
return;
```

File: server.js [hunk 2]
```diff
@@ -2113,6 +2113,54 @@ cache(function(data, match, sendBadge, request) {
...
+  // http://developer.github.com/v3/#user-agent-required
+  // https://developer.github.com/v3/licenses/
+  var customHeaders = {
+    'User-Agent': 'Shields.io',
+    'Accept': 'application/vnd.github.drax-preview+json'
+  };
+  request(apiUrl, { headers: customHeaders }, function(err, res, buffer) {
+    if (err != null) {
+      badgeData.text[1] = 'inaccessible';
+      sendBadge(format, badgeData);
```
Comment (espadrine) [inline server.js]: Please add a return statement here.

**Comment by espadrine** [comment]
Great work! Can't wait to merge this.

**Comment by montanaflynn** [comment]
Made the changes, I thought that the text should be updated to differentiate between a not found repo and unknown license, but happy to hear feedback on that as well.

**Comment by espadrine** [comment]
Thanks a lot! I'll put it live in 24h.

**Pull request merged**